README.md

```diff
@@ -1,289 +1,590 @@
+
+The contents of a secret entry may be anything representable as bytes on the file system including, but not limited to, a literal string value (e.g. `db-password`), a language-specific binary (e.g. a Java `KeyStore` with a private key and X.509 certificate), or an indirect pointer to another system for value resolution (e.g. `vault://production-database/password`).
+
+The collection of files within the directory **MAY** change between container launches.  The collection of files within the directory **MUST NOT** change during the lifetime of the container.
+
+## Example Directory Structure
+
+```plain
+$SERVICE_BINDING_ROOT
+├── account-database
+│   ├── kind
+│   ├── connection-count
+│   ├── uri
+│   ├── username
+│   └── password
+└── transaction-event-stream
+│   ├── kind
+│   ├── connection-count
+│   ├── uri
+│   ├── certificates
+│   └── private-key
+```
+
+# Service Binding
+
+A Service Binding describes the connection between a [Provisioned Service](#provisioned-service) and an [Application Projection](#application-projection).  It is codified as a concrete resource type.  Multiple Service Bindings can refer to the same service.  Multiple Service Bindings can refer to the same application.
+
+A Service Binding resource **MUST** define a `.spec.application` which is an `ObjectReference` to a `PodSpec`-able resource.  A Service Binding resource **MUST** define a `.spec.service` which is an `ObjectReference` to a Provisioned Service-able resource.  A Service Binding resource **MAY** define a `.spec.name` which is the name of the service when projected into to the application.
+
+A Service Binding resource **MUST** define a `.status.conditions` which is an array of `Condition` objects.  A `Condition` object **MUST** define `type`, `status`, and `lastTransitionTime` entries.  At least one condition containing a `type` of `Ready` must be defined.  The `status` of the `Ready` condition **MUST** have a value of `True`, `False`, or `Unknown`.  The `lastTranstionTime` **MUST** contain the last time that the condition transitioned from one status to another.  A Service Binding resource **MAY** define `reason` and `message` entries to describe the last `status` transition.
+
+## Resource Type Schema
+
+```yaml
 apiVersion: service.binding/v1alpha1
 kind: ServiceBinding
 metadata:
-  name: example-service-binding
+  name:                 # string
 spec:
-  mountPathPrefix: "/var/bindings"
-  services:
-    - group: postgres.dev
-      kind: Service
-      name: global-user-db
-      version: v1beta1
-      id: postgres-global-user
-    - group: ibmcloud.ibm.com
-      version: v1alpha1
-      kind: Binding
-      name: coligo-service-binding
-      id: coligo-service-binding
-  application:
-    name: nodejs-rest-http-crud
-    group: apps
-    version: v1
-    resource: deployments
+  name:                 # string, optional, default: .metadata.name
+  kind:                 # string, optional
```

Comment:
I find it confusing that "kind" is used for classification of the Provisioned Service (and represented as an entry within the Secret) since actual ObjectReferences are involved here as well (application and service) for which "kind" maps to the actual K8s resource `Kind`.

Reply:
We could rename it to `type` to avoid the overloading

Reply:
I agree.  Users may be thinking that the `kind` they put here should always match the k8s kind of the service they are pointing to.

Reply:
Well, I don't think that that is necessarily a bad thing.  I suspect that in many cases that the resource kind _is_ a good choice for the `Secret` `kind`.  In fact, there was a draft of this part of the spec that defined `.spec.service.kind` as the default if you hadn't explicitly set it, and it wasn't already in the `Secret.  I ended up dropping it for the same reason as `.metadata.name -> .spec.service.name -> .spec.name`; a bit too much complexity.

I'm open to changing it, but I do genuinely like idea of resource kinds being a clear analog.

Reply:
Addressed in e9d77e8